frontend/src/metabase/admin/settings/components/Email/SMTPConnectionForm.tsx

```diff
@@ -58,54 +68,56 @@ const anySchema = Yup.mixed().nullable().default(null);
-const getFormValueSchema = (elementMap: _.Dictionary<SettingElement>) => {
+const getFormValueSchema = (
+  settingsDetails: SettingDefinitionMap | undefined,
+) => {
   return Yup.object({
-    "email-smtp-host": elementMap["email-smtp-host"].is_env_setting
+    "email-smtp-host": settingsDetails?.["email-smtp-host"].is_env_setting
       ? anySchema
       : Yup.string().required(Errors.required).default(""),
-    "email-smtp-port": elementMap["email-smtp-port"].is_env_setting
+    "email-smtp-port": settingsDetails?.["email-smtp-port"].is_env_setting
       ? anySchema
       : Yup.number()
           .positive()
           .nullable()
           .required(Errors.required)
           .default(null),
-    "email-smtp-security": elementMap["email-smtp-security"].is_env_setting
+    "email-smtp-security": settingsDetails?.["email-smtp-security"]
+      .is_env_setting
       ? anySchema
       : Yup.string().default("none"),
-    "email-smtp-username": elementMap["email-smtp-username"].is_env_setting
+    "email-smtp-username": settingsDetails?.["email-smtp-username"]
+      .is_env_setting
       ? anySchema
       : Yup.string().default(""),
-    "email-smtp-password": elementMap["email-smtp-password"].is_env_setting
+    "email-smtp-password": settingsDetails?.["email-smtp-password"]
+      .is_env_setting
       ? anySchema
       : Yup.string().default(""),
   });
 };
 
-export const SMTPConnectionForm = ({
-  elements,
-  settingValues,
-}: SMTPConnectionFormProps) => {
+export const SMTPConnectionForm = () => {
   const [sendingEmail, setSendingEmail] = useState<ButtonStateType>("default");
   const [testEmailError, setTestEmailError] = useState<string | null>(null);
-
+  const { data: settingsDetails } = useGetAdminSettingsDetailsQuery();
   const isHosted = useSelector(getIsHosted);
   const isEmailConfigured = useSelector(getIsEmailConfigured);
   const dispatch = useDispatch();
 
-  const elementMap = useMemo(() => _.indexBy(elements, "key"), [elements]);
-
   const initialValues = useMemo<FormValueProps>(
     () => ({
-      "email-smtp-host": settingValues["email-smtp-host"],
-      "email-smtp-port": settingValues["email-smtp-port"],
-      "email-smtp-security": settingValues["email-smtp-security"] ?? "none",
-      "email-smtp-username": settingValues["email-smtp-username"] ?? "",
-      "email-smtp-password": settingValues["email-smtp-password"] ?? "",
+      "email-smtp-host": settingsDetails?.["email-smtp-host"].value ?? null,
+      "email-smtp-port": settingsDetails?.["email-smtp-port"].value ?? null,
+      "email-smtp-security":
+        settingsDetails?.["email-smtp-security"].value ?? "none",
+      "email-smtp-username":
+        settingsDetails?.["email-smtp-username"].value ?? "",
+      "email-smtp-password":
+        settingsDetails?.["email-smtp-password"].value ?? "",
     }),
-    [settingValues],
+    [settingsDetails],
   );
-
   const handleClearEmailSettings = useCallback(async () => {
```

Comment:
clearing email settings doesn't reset the form for me, not sure why 🤔

Reply:
This seems to work now the values are hooked up with rtk hooks